src/components/shared/misc/SummarisedMultiSelector.vue

```diff
@@ -0,0 +1,247 @@
+                        value: 'id',
+                    }
+                },
+            },
+
+            /**
+            * The current value of the dropdown, v-model alias
+            */
+            value: {
+                required: true,
+                type: [String, Array],
+                default() {
+                    return ''
+                },
+            },
+
+            /**
+            * An optional placeholder for the dropdown
+            */
+            placeholder: {
+                type: String,
+                default() {
+                    return ''
+                },
+            },
+
+            /**
+            * Default text for the dropdown summary popup
+            */
+            defaultSummary: {
+                type: String,
+                default() {
+                    return 'Select an item'
+                },
+            },
+
+            /**
+            * Which field in the options to use in the summary popup
+            */
+            summaryField: {
+                type: String,
+                default() {
+                    return 'name'
+                },
+            },
+
+            /**
+            * Semantic ui dropdown configuration
+            */
+            dropdownSettings: {
+                type: Object,
+                default() {
+                    return {
+                        useLabels: false,
+                        showOnFocus: true,
+                        onHide: () => this.$emit('dropdown-hidden'),
+                    }
+                },
+            },
+
+            /**
+            * If the component should be in read only mode
+            */
+            readOnly: {
+                type: Boolean,
+                default() {
+                    return false
+                },
+            },
+
+            /**
+            * If the component should be inline
+            */
+            inline: {
+                type: Boolean,
+                default() {
+                    return true
+                },
+            },
+
+            /**
+            * If the component should be fluid
+            */
+            fluid: {
+                type: Boolean,
+                default() {
+                    return true
+                },
+            },
+        },
+
+        computed: {
+            model: {
+                get() {
+                    return this.value
+                },
+
+                set(val) {
+                    this.$emit('input', val)
+                },
+            },
+
+            computedModel() {
+                return Array.isArray(this.model) ? this.model : this.model.split(',')
+            },
+
+            selectedLength() {
+                return this.computedModel.length
+            },
+
+            selectedItems() {
+                return this.options.filter((option) => {
+                    if (this.computedModel.toString().indexOf(option[this.fields.value].toString()) > -1) {
```

Comment:
This logic is a bit iffy, because we are casting the array to a string and then searching for substrings this will return true for substrings of the selected id. 

For example consider the following options structure:
```js
[
  { id: 1, name: 'One' },
  { id: 2, name: 'Two' },
  { id: 12, name: 'Twelve' },
],
```
Selecting `Twelve` will result in all three options being shown in the summary because of the following checks
```js
'12'.indexOf('1') // 0
'12'.indexOf('2') // 1
'12'.indexOf('12') // 0
```

I will probably be releasing a PR later this morning to add some more test coverage for this and to refactor a few things, I will look at tidying this up too.

Reply:
Thanks for explaining this, I will take note on the changes you make. Cheers